From 65c1b7d52d1b647a9ebccb61a7665e5562e4acbc Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Tue, 21 Oct 2014 13:26:43 -0400
Subject: [fs] ext4: remove unused code from ext4_remove_blocks()

Message-id: <1413898007-15450-12-git-send-email-lczerner@redhat.com>
Patchwork-id: 97884
O-Subject: [RHEL 7.1 PATCH v2 11/15] ext4: remove unused code from ext4_remove_blocks()
Bugzilla: 1144326
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

BZ 1144326
https://bugzilla.redhat.com/show_bug.cgi?id=1144326

BREW 8095073
https://brewweb.devel.redhat.com/taskinfo?taskID=8095073

Upstream commit 78fb9cdf035d88a39a5e5f83bb8788e4fe7c1f72

 The "head removal" branch in the condition is never used in any code
 path in ext4 since the function only caller ext4_ext_rm_leaf() will make
 sure that the extent is properly split before removing blocks. Note that
 there is a bug in this branch anyway.

 This commit removes the unused code completely and makes use of
 ext4_error() instead of printk if dubious range is provided.

 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
 Reviewed-by: Jan Kara <jack@suse.cz>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/extents.c b/fs/ext4/extents.c
index e58c83f..0dab624 100644
--- a/fs/ext4/extents.c
+++ b/fs/ext4/extents.c
@@ -2433,23 +2433,10 @@ static int ext4_remove_blocks(handle_t *handle, struct inode *inode,
    *partial_cluster = EXT4_B2C(sbi, pblk);
   else
    *partial_cluster = 0;
- } else if (from == le32_to_cpu(ex->ee_block)
-     && to <= le32_to_cpu(ex->ee_block) + ee_len - 1) {
-  /* head removal */
-  ext4_lblk_t num;
-  ext4_fsblk_t start;
-
-  num = to - from;
-  start = ext4_ext_pblock(ex);
-
-  ext_debug("free first %u blocks starting %llu\n", num, start);
-  ext4_free_blocks(handle, inode, NULL, start, num, flags);
-
- } else {
-  printk(KERN_INFO "strange request: removal(2) "
-    "%u-%u from %u:%u\n",
-    from, to, le32_to_cpu(ex->ee_block), ee_len);
- }
+ } else
+  ext4_error(sbi->s_sb, "strange request: removal(2) "
+      "%u-%u from %u:%u\n",
+      from, to, le32_to_cpu(ex->ee_block), ee_len);
  return 0;
 }
 
-- 
1.7.1